2.5/README.md

```diff
@@ -23,26 +23,111 @@ modules for their web applications. There is no guarantee for any specific npm o
 version, that is included in the image; those versions can be changed anytime and
 the nodejs itself is included just to make the npm work.
 
-Usage
+Usage in Openshift
 ---------------------
-For this, we will assume that you are using the `rhscl/ruby-25-rhel7 image`, available via `ruby:2.5` imagestream tag in Openshift.
+For this, we will assume that you are using the `ubi8/ruby-25 image`, available via `ruby:2.5` imagestream tag in Openshift.
 Building a simple [ruby-sample-app](https://github.com/sclorg/s2i-ruby-container/tree/master/2.5/test/puma-test-app) application
 in Openshift can be achieved with the following step:
 
     ```
-    oc new-app ruby:2.5~https://github.com/sclorg/s2i-ruby-container.git --context-dir=2.5/test/puma-test-app/
+    oc new-app ruby:2.5~https://github.com/sclorg/rails-ex.git
     ```
 
-The same application can also be built using the standalone [S2I](https://github.com/openshift/source-to-image) application on systems that have it available:
+**Accessing the application:**
+```
+$ oc get pods
+$ oc exec <pod> -- curl 127.0.0.1:8080
+```
 
-    ```
-    $ s2i build https://github.com/sclorg/s2i-ruby-container.git --context-dir=2.5/test/puma-test-app/ rhscl/ruby-25-rhel7 ruby-sample-app
-    ```
+Source-to-Image framework and scripts
+-------------------------------------
+This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/3.11/creating_images/s2i.html)
+(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
+which makes it easy to write images that take application source code as
+an input, use a builder image like this Node.js container image, and produce
+a new image that runs the assembled application as an output.
 
-**Accessing the application:**
+To support the Source-to-Image framework, important scripts are included in the builder image:
```

Comment:
Isn't this really general knowledge for s2i, to create the container, but not needed to use one, as we supply those scripts?

I'm baffled, why would users change this? Or is this meant as a documentation for the images? shouldn't this be on the upper level in https://github.com/sclorg/s2i-ruby-container/blob/master/README.md?

Reply:
From my point of view, this part of README says how to build own container app using s2i scripts or using Dockerfile from ruby 2.5 container.

Reply:
Yes, you're right, that's why I think this content does not belong here as it's not related to container usage, but to container creation instead.

Reply:
I would say building a new image on top of our own using the s2i technology is definitely a way (one way) of using the image.
Note that the user can very well want to override any of these scripts.

Reply:
Right, which I've done - for s2i, but we don't say "how". Best practises, like prefixing the script... anyway, I'd label it advanced usage and put it at bottom at least.